crates/sui-framework/packages/bridge/sources/bridge.move

```diff
@@ -349,13 +357,292 @@ module bridge::bridge {
         (option::some(token), owner)
     }
 
-    fun next_seq_num(self: &mut BridgeInner, msg_type: u8): u64 {
+    public fun execute_system_message(
+        self: &mut Bridge,
+        message: BridgeMessage,
+        signatures: vector<vector<u8>>,
+    ) {
+        let message_type = message::message_type(&message);
+        // TODO: test version mismatch
+        assert!(message::message_version(&message) == ACCEPTED_MESSAGE_VERSION, EUnexpectedMessageVersion);
+        let inner = load_inner_mut(self);
+
+        assert!(message::source_chain(&message) == inner.chain_id, EUnexpectedChainID);
+
+        // check system ops seq number and increment it
+        let expected_seq_num = get_current_seq_num_and_increment(inner, message_type);
+        assert!(message::seq_num(&message) == expected_seq_num, EUnexpectedSeqNum);
```

Comment:
If I understand this correctly, all operations need to be executed in a serial order from now on?

Reply:
Discussed on the meet, turns out I missread things. That's only for system messages anyways!